analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -146,4 +146,67 @@ public static ProgramState LearnFrom(ProgramState state, IPropertyReferenceOpera
         }
         return state;
     }
+
+    public static ProgramState LearnFrom(ProgramState state, IInvocationOperationWrapper invocation)
```

Comment:
The methods `ObjectCreationConstraint` and `ArrayCreationConstraint` should be adapted to the new `LearnFrom` format.